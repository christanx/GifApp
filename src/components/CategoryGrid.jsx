import useFetchGif from '../hooks/useFetchGif';
import CategoryGridItem from './CategoryGridItem';
import PropTypes from 'prop-types'

const CategoryGrid = ({ category }) => {
    const { data: images, loading } = useFetchGif(category);

    return (
        <>
            <h3 className='animate__animated animate__fadeInDown'>{category}</h3>
            {
                loading && <span>loading...</span>
            }
            <div className='card-grid'>
                {
                    images.map(img => (
                        <CategoryGridItem key={img.id} {...img} />
                    ))
                }
            </div>
        </>
    )
}

CategoryGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default CategoryGrid
