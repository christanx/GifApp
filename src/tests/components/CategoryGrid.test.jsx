import '@testing-library/jest-dom';
import { shallow } from "enzyme"
import CategoryGrid from "../../components/CategoryGrid"
import useFetchGif from "../../hooks/useFetchGif";
jest.mock("../../hooks/useFetchGif");


describe('Testing <CategoryGrid />', () => {
    const category = 'Goku';

    test('Component successful', () => {
        useFetchGif.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<CategoryGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('El useFetchGif debe retornar items', () => {
        const gifs = [
            {
                id: '01',
                url: 'http://localhost',
                title: 'testing'
            },
            {
                id: '02',
                url: 'http://localhost',
                title: 'testing'
            }
        ]
        
        useFetchGif.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<CategoryGrid category={category} />);
        expect(wrapper).toMatchSnapshot();

        const boxLoading = wrapper.find('span').exists();
        expect(boxLoading).toBe(false);

        const gridItems = wrapper.find('CategoryGridItem');
        expect(gridItems.length).toBe(gifs.length)
    })
    
    
})
