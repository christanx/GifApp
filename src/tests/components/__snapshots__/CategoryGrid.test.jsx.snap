// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CategoryGrid /> Component successful 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Goku
  </h3>
  <span>
    loading...
  </span>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing <CategoryGrid /> El useFetchGif debe retornar items 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <CategoryGridItem
      id="01"
      key="01"
      title="testing"
      url="http://localhost"
    />
    <CategoryGridItem
      id="02"
      key="02"
      title="testing"
      url="http://localhost"
    />
  </div>
</Fragment>
`;
